beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -474,12 +480,71 @@ func (s *Server) GetAttesterSlashings(w http.ResponseWriter, r *http.Request) {
 			ss = append(ss, s)
 		} else {
 			httputil.HandleError(w, fmt.Sprintf("unable to convert slashing of type %T", slashing), http.StatusInternalServerError)
-			return
+			return "", nil
 		}
 	}
 	slashings := structs.AttesterSlashingsFromConsensus(ss)
+	return v, slashings
+}
 
-	httputil.WriteJson(w, &structs.GetAttesterSlashingsResponse{Data: slashings})
+// GetAttesterSlashingsV2 retrieves attester slashings known by the node but
+// not necessarily incorporated into any block, supporting both AttesterSlashing and AttesterSlashingElectra.
+func (s *Server) GetAttesterSlashingsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashingsV2")
+	defer span.End()
+
+	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
+	if err != nil {
+		httputil.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	v := headState.Version()
+	resp := &structs.GetAttesterSlashingsV2Response{Version: version.String(v)}
+
+	sourceSlashings := s.SlashingsPool.PendingAttesterSlashings(ctx, headState, true /* return unlimited slashings */)
+
+	if v >= version.Electra {
+		// Initialize the slice with length equal to len(sourceSlashings) so that we can index into it
+		ss := make([]*eth.AttesterSlashingElectra, len(sourceSlashings))
+		for i, slashing := range sourceSlashings {
+			a, ok := slashing.(*eth.AttesterSlashingElectra)
+			if ok {
+				ss[i] = a // Safe to index here because the slice has the correct length
+			} else {
+				httputil.HandleError(w, fmt.Sprintf("unable to convert electra slashing of type %T", slashing), http.StatusInternalServerError)
+				return
+			}
+		}
+		for _, slashing := range ss {
```

Comment:
why not just combine this? I think you can avoid this loop by just marshalling and adding to the response data in the above loop